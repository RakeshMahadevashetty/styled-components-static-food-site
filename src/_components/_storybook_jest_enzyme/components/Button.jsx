import propTypes from "prop-types";

const Button = ({ label, backgroundColor = "red", size = "md", handleClick }) => {
  let scale = 1;
  
  if (size === "sm") scale = 0.75;
  if (size === "lg") scale = 1.5;

  const style = {
    backgroundColor,
    padding: `${scale * 0.5}rem ${scale * 1}rem`,
    border: "none",
    margin: "1rem"
  };
  return (
    <button onClick={handleClick} style={style} data-testid="button">
      {label}
    </button>
  );
};

Button.propTypes = {
  label: propTypes.string,
  backgroundColor: propTypes.string,
  size: propTypes.oneOf(["sm", "md", "lg"]),
  handleClick: propTypes.func,
};

export default Button;
