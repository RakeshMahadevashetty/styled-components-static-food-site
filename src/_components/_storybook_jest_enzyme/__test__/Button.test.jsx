import { render } from "@testing-library/react";
import { shallow } from "enzyme";
import Button from "../components/Button";

describe("Button component", () => {
  it("Rendered Button", () => {
    const { getByTestId } = render(<Button label="Click" />);
    const currentButton = getByTestId("button");
    expect(currentButton).toBeTruthy();
  });
});

test("Should test Button component", () => {
  const wrapper = shallow(<Button label="click me.." size="sm"/>);
  expect(wrapper).toMatchSnapshot();
});

test("Should call handleClick method", () => {
  const wrapper = shallow(
    <Button
      label={"test click"}
      handleClick={() => console.log("Button clicked")}
      size="lg"
    />
  );
  const instance = wrapper.instance();
  // instance.handleClick({
  //   preventDefault: () => {}
  // })
});
