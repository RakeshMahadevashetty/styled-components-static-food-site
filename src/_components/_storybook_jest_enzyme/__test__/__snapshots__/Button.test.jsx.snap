// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should test Button component 1`] = `
<button
  data-testid="button"
  style={
    Object {
      "backgroundColor": "red",
      "border": "none",
      "margin": "1rem",
      "padding": "0.375rem 0.75rem",
    }
  }
>
  click me..
</button>
`;
